tests/server/client.go

```diff
@@ -0,0 +1,103 @@
+package testsuite
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	abcicli "github.com/tendermint/abci/client"
+	"github.com/tendermint/abci/types"
+	crypto "github.com/tendermint/go-crypto"
+	cmn "github.com/tendermint/tmlibs/common"
+)
+
+func InitChain(client abcicli.Client) error {
+	total := 10
+	vals := make([]*types.Validator, total)
+	for i := 0; i < total; i++ {
+		pubkey := crypto.GenPrivKeyEd25519FromSecret([]byte(cmn.Fmt("test%d", i))).PubKey().Bytes()
+		power := cmn.RandInt()
+		vals[i] = &types.Validator{pubkey, uint64(power)}
+	}
+	err := client.InitChainSync(types.RequestInitChain{Validators: vals})
+	if err != nil {
+		fmt.Println("Failed test: InitChain - %v", err)
+		return err
+	}
+	fmt.Println("Passed test: InitChain")
+	return nil
+}
+
+func SetOption(client abcicli.Client, key, value string) error {
+	res := client.SetOptionSync(key, value)
+	_, _, log := res.Code, res.Data, res.Log
+	if res.IsErr() {
+		fmt.Println("Failed test: SetOption")
+		fmt.Printf("setting %v=%v: \nlog: %v", key, value, log)
+		fmt.Println("Failed test: SetOption")
+		return errors.New(res.Error())
+	}
+	fmt.Println("Passed test: SetOption")
+	return nil
+}
+
+func Commit(client abcicli.Client, hashExp []byte) error {
+	res := client.CommitSync()
+	_, data, log := res.Code, res.Data, res.Log
+	if res.IsErr() {
+		fmt.Println("Failed test: Commit")
+		fmt.Printf("committing %v\nlog: %v", log)
+		return errors.New(res.Error())
+	}
+	if !bytes.Equal(data, hashExp) {
+		fmt.Println("Failed test: Commit")
+		fmt.Printf("Commit hash was unexpected. Got %X expected %X",
```

Comment:
If this test fails, the first value is incorrectly rendered. For example, if the second value is `0000000000000005` and the first one is the same, but with 0 byte added at the end, we get “Got 303030303030303030303030303030353030 expected 0000000000000005”. Maybe it has something to do with the fact that the first value is `data.Bytes`, while the second is `[]byte`?

Reply:
have you tried `data.Bytes()`?

Reply:
`data.Bytes()` helps, thanks.